core/drivers/stm32_bsec.c

```diff
@@ -430,14 +423,13 @@ TEE_Result stm32_bsec_permanent_lock_otp(uint32_t otp_id)
 TEE_Result stm32_bsec_write_debug_conf(uint32_t value)
 {
 	TEE_Result result = TEE_ERROR_GENERIC;
-	uint32_t masked_val = value & BSEC_DEN_ALL_MSK;
```

Comment:
Previous code prevented caller from writting into reserved bits (that should remain to 0 as per ref manual). I think we should rpeserved the mask or at least assert that `value` sets only valid bits (`assert(!(value & ~BSEC_DEN_ALL_MSK));`).

Reply:
Actually, the BSEC_DEN_ALL_MSK isn't correct atm as it covers platform-dependant reserved bits that shouldn't be set (Bit 8 and 0 for STM32MP13x and Bit 0 for STM32MP15x).
A way of doing it would be to create a specific mask for STM32MP13x and a different one for 15x platforms. But then, i

The check is fully removed here to simplify code and transfer the responsibility to check what has been written in this register to the caller of the function. Now, it should be managed by caller with a sequence read/modify/write to avoid update of reserved bits

Reply:
Ok. Fine it is caller responsibility.
`Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>` for commit "drivers: stm32_bsec: move debug defines to include".